src/tools/r2rdump/R2RDump.cs

```diff
@@ -4,55 +4,436 @@
+        private bool _help = false;
+        private IReadOnlyList<string> _inputFilenames = Array.Empty<string>();
+        private string _outputFilename = null;
+        private bool _raw = false;
+        private bool _header = false;
+        private bool _disasm = false;
+        private IReadOnlyList<string> _queries = Array.Empty<string>();
+        private IReadOnlyList<string> _keywords = Array.Empty<string>();
+        private IReadOnlyList<int> _runtimeFunctions = Array.Empty<int>();
+        private IReadOnlyList<string> _sections = Array.Empty<string>();
+        private bool _diff = false;
+
+        private R2RDump()
+        {
+        }
+
+        private ArgumentSyntax ParseCommandLine(string[] args)
+        {
+            ArgumentSyntax argSyntax = ArgumentSyntax.Parse(args, syntax =>
+            {
+                syntax.ApplicationName = "R2RDump";
+                syntax.HandleHelp = false;
+                syntax.HandleErrors = false;
+
+                syntax.DefineOption("h|help", ref _help, "Help message for R2RDump");
+                syntax.DefineOptionList("i|in", ref _inputFilenames, "Input file(s) to dump. Expects them to by ReadyToRun images");
+                syntax.DefineOption("o|out", ref _outputFilename, "Output file path. Dumps everything to the specified file except help message and exception messages");
+                syntax.DefineOption("v|verbose|raw", ref _raw, "Dump the raw bytes of each section or runtime function");
+                syntax.DefineOption("l|less", ref _header, "Only dump R2R header");
+                syntax.DefineOption("d|disasm", ref _disasm, "Show disassembly of methods or runtime functions");
+                syntax.DefineOptionList("q|query", ref _queries, "Query method by exact name, signature, row id or token");
+                syntax.DefineOptionList("k|keyword", ref _keywords, "Search method by keyword");
+                syntax.DefineOptionList("r|rtf", ref _runtimeFunctions, ArgStringToInt, "Get one runtime function by id or relative virtual address");
+                syntax.DefineOptionList("s|section", ref _sections, "Get section by keyword");
+                syntax.DefineOption("diff", ref _diff, "Compare two R2R images"); // not yet implemented
+            });
+
+            return argSyntax;
+        }
+
+        private int ArgStringToInt(string arg)
+        {
+            bool isNum;
+            return ArgStringToInt(arg, out isNum);
+        }
+
+        /// <summary>
+        /// Converts string passed as cmd line args into int, works for hexidecimal with 0x as prefix
+        /// </summary>
+        /// <param name="arg">The argument string to convert</param>
+        /// <param name="isNum">Set to false if conversion failed</param>
+        private int ArgStringToInt(string arg, out bool isNum)
+        {
+            int n = -1;
+            if (arg.StartsWith("0x", StringComparison.OrdinalIgnoreCase))
```

Comment:
This could possibly fail if there was leading whitespace in the string argument. To make this a little more resilient, you can first save `arg.TrimStart()` to a string variable, that way you know there is no leading whitespace, and then continue operation on the new variable.